src/base/perf_monitor.h

```diff
@@ -1375,30 +1375,31 @@ extern bool set_diag_value (T_DIAG_OBJ_TYPE type, int value, T_DIAG_VALUE_SETTYP
 #endif /* DIAG_DEVEL */
 
 #ifndef DIFF_TIMEVAL
-#define DIFF_TIMEVAL(start, end, elapsed) \
-    do { \
-      (elapsed).tv_sec = (end).tv_sec - (start).tv_sec; \
+#define DIFF_TIMEVAL(start, end, elapsed)                  \
```

Comment:
On the principle you touched it, you fix it, please replace with inline function :):

```c
STATIC_INLINE perfmon_diff_timeval (...
...
}
#define DIFF_TIMEVAL perfmon_diff_timeval
```

Same for ADD_TIMEVAL.